Celeste.Mod.mm/Mod/Meta/MapMeta.cs

```diff
@@ -132,9 +133,83 @@ public class MapMeta : IMeta {
                     Modes[i] = null;
                 }
             }
+            BinaryPacker.Element modeMeta = meta.Children?.FirstOrDefault(el => el.Name == "mode");
+            if (modeMeta != null) {
+                Mode = new MapMetaModeProperties(modeMeta);
+            }
+        }
+
+        public void AddTo(MapMeta other) {
```

Comment:
AddTo is the main utility method I add - it does the merging between two MapMetas. It mutates other, overriding any properties which are set by `this`.